### PR TITLE
Mesh network management API

### PR DESCRIPTION
**Pull request by sergeuz**
This PR implements a set of API functions for mesh network management.

### CODE REVIEW

**Comment by sergeuz** [comment]
[ch28309]

**Comment by stefan-moraru** [comment]
Nice work @sergeuz. Good to see you in JavaScript land! I noticed a few `TODO`s in your code, are those issues we want to address now, or after shipping?

File: src/Particle.js [hunk 1]
```diff
@@ -1329,6 +1329,109 @@ class Particle {
...
+
+	/**
+	 * List all mesh networks
+	 * @param {Object} options Options for this API call
+	 * @param {String} options.auth Access token
+	 * @param {Object} [options.context] Request context
+	 * @return {Promise<Object>}
+	 */
+	listMeshNetworks({ auth, context }) {
+		// TODO: Pagination
```
Comment (monkbroc) [inline src/Particle.js]: Pagination is supported with `?page=1&per_page=1`

File: src/Particle.js [hunk 2]
```diff
@@ -1329,6 +1329,109 @@ class Particle {
...
+	/**
+	 * List all devices of a mesh network.
+	 * @param {Object} options Options for this API call
+	 * @param {String} options.networkId Network ID or name
+	 * @param {String} options.auth Access token
+	 * @param {Object} [options.context] Request context
+	 * @return {Promise<Object>}
+	 */
+	listMeshNetworkDevices({ networkId, auth, context }) {
+		// TODO: Pagination and filtering
```
Comment (monkbroc) [inline src/Particle.js]: Filtering and pagination is supported through `?page=2&per_page=1&role=node`

**Comment by sergeuz** [comment]
@monkbroc Added parameters for pagination and filtering by device type: https://github.com/particle-iot/particle-api-js/pull/98/commits/4ada7d4ca7ba504fec406e653bb828cae4e64411

**Comment by monkbroc** [comment]
Go ahead and merge when you are ready

**Pull request merged**